packages/core/src/components/forms/_file-upload.scss

```diff
@@ -135,3 +118,22 @@ $file-upload-button-width-large: $pt-grid-size * 8.5;
     }
   }
 }
+
+.pt-file-upload-input {
+  @include pt-input();
+  @include overflow-ellipsis();
+  position: absolute;
+  top: 0;
+  right: 0;
+  left: 0;
+  height: $pt-input-height;
+  padding-right: $file-upload-button-width + $input-padding-horizontal;
+  line-height: $pt-input-height;
+  color: $pt-text-color-muted;
+  user-select: none;
+
+  .pt-dark & {
```

Comment:
add `.pt-large & { ?? }` support. there's a mixin for it, check original input styles.